model-optimizer/extensions/ops/upsample.py

```diff
@@ -63,5 +63,6 @@ def upsample_infer(node: Node):
                                                      width=out_width)
         else:
             assert node.in_node(1).value is not None
+            eps = 1e-5  # This is to make rounding in case of very close number to round to closest instead of down
             # generic output shape calculation to support 5D input shape case
-            node.out_node().shape = np.array(input_shape * node.in_node(1).value).astype(np.int64)
+            node.out_node().shape = np.array((input_shape + eps) * node.in_node(1).value).astype(np.int64)
```

Comment:
Why?

Reply:
There is a very common problem in PyTorch. ONNX spec states that shape rounding should be done using `floor` function, but PyTorch creates such scales in Resize-10 that create very close to desired number, e.g. 550.99999, but which will be rounded to 350 instead of 351, my fix resolves the problem.

Reply:
Well. But is it correct for ONNX but not PyTorch model? And UpsampleOp is also used in the transformation BatchToSpaceToUpsample. Is your fix correct for this transformation?

Reply:
We had a discussion that we need to understand first how ONNX runtime handles this case and update MO/IE code appropriately.